### ISSUE TITLE
TextField required: Property "required" does not exist

### ISSUE THREAD

**Issue by LarsLiden** [OfficeDev/office-ui-fabric-react#2396]
### Bug Report

- __Package version(s)__:^4.23.0

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (No)

Requested priority: (Normal) 

Products/sites affected: (if applicable)
MS BLIS

#### Describe the issue:
Following the example in the documentation (see below) and getting the error shown above

`                        <TextField
                            required={ true }
                            onChanged={this.contentChanged.bind(this)}
                            label="Payload"
                            placeholder="Payload..."
                            value={this.state.contentVal} />`

#### Actual behavior:

getting the error shown above (Property "required" does not exist)

#### Expected behavior:

should work like the sample

**Comment by micahgodbolt** [comment]
Have you tried this with 4.31.0? We had a rough patch with typings mismatch. There shouldn't be anything wrong with the textfield required property.

**Comment by pchc2005** [comment]
Hi @micahgodbolt , 

I have met this problem as well, and my version of office-ui-fabric-react is 4.31.0.

**Comment by pchc2005** [comment]
not only 'required'. This problem is also happened to 'name' and 'placeholder'.

**Comment by enriquin** [comment]
In my case that was only an intellisense problem, since visual studio was using Typescript 2.2 for the intellisense, but wepack was really using the 2.4 version for the real transpiling process. All the properties that showed errors where defined in "d.ts" files from the 2.2 version.

The problem arised a few days ago (when the typing changed?). Then I realized I was still using TS 2.2, then I:

1. Installed latest typescript version.
2. Changed Visual Studio's intellisense settings as explained here: https://github.com/Microsoft/TypeScript/wiki/Updating-TypeScript-in-Visual-Studio-2017 and got not more intellisense problems.

Hope it helps!
